### PR TITLE
Don't pass `-fno-inline-functions` when `-Oz`/`-Os` are used

### PR DESCRIPTION
**Pull request by sbc100**
We have no reason not to let clang use its own heuristics here.

As a followup we can consider completely removing the `INLINGING_LIMIT` setting since it no longer serves any purpose other than to set this clang flag.

### CODE REVIEW

**Comment by kripken** [comment]
This does seem like the right thing to me. However reading the code size changes my impression is the opposite of yours - seems to be more regressions than wins, and some seem significant (e.g. `metadce_cxx_noexcept` and `random_printf`). Perhaps it's worth looking into those a little more - might be system libs we should use `-Oz` over `-Os`, or even `-Oz -fno-inline-functions`?

**Comment by sbc100** [comment]
Adding back `-fno-inline-functions` to system library builds to avoid any code size regressions.

**Review by kripken — APPROVED** [review]
Thanks, lgtm.

If one of us has time eventually it might be interesting to look into the system lib flags and check if they are optimal. But seems low priority to me.

File: ChangeLog.md [hunk 1]
```diff
@@ -20,6 +20,9 @@ See docs/process.md for more on how version tagging works.
 
 3.1.22 (in development)
 -----------------------
+- Using `-Oz` or `-Os` will no longer pass `-fno-inline-functions` to clang and
+  instead rely on clang's normal inline hueristics for these optimization
```
Comment (aheejin) [inline ChangeLog.md:24]:
```suggestion
  instead rely on clang's normal inline heuristics for these optimization
```

File: tools/system_libs.py [hunk 2]
```diff
@@ -713,7 +713,7 @@ class libc(MuslInternalLibrary,
   # functions to something else based on assumptions that they behave exactly
   # like the standard library. This can cause unexpected bugs when we use our
   # custom standard library. The same for other libc/libm builds.
-  cflags = ['-Os', '-fno-builtin']
+  cflags = ['-Os', '-fno-inline-functions', '-fno-builtin']
```
Comment (aheejin) [inline tools/system_libs.py:716]: Maybe one-line comment on why `-fno-inline-functions` is good for libraries?

**Review by aheejin — APPROVED** [review]

**Pull request merged**